src/Workspaces/Core/Portable/Workspace/Workspace_Events.cs

```diff
@@ -168,6 +170,72 @@ protected Task RaiseDocumentClosedEventAsync(Document document)
             }
         }
 
+        /// <summary>
+        /// An event that is fired when a documents is opened in the editor.
+        /// </summary>
+        internal event EventHandler<DocumentEventArgs> SourceGeneratedDocumentOpened
```

Comment:
📝 This is the trigger for solution crawler to run. I could use `DocumentOpened`, but then we'd need to audit all consumers of this public API to make sure they are capable of handling events where a subsequent `GetDocument` call returns null.

Reply:
⏱️ Am combining these events

Reply:
➡️ These events are now combined